mlir/include/mlir/Dialect/Linalg/Passes.td

```diff
@@ -141,4 +141,63 @@ def LinalgDetensorizePass : InterfacePass<"linalg-detensorize", "FunctionOpInter
   ];
 }
 
+def LinalgBlockPackMatmul : Pass<"linalg-block-pack-matmul"> {
```

Comment:
> Such passes in core do not age well.

That's true, but I think it's orthogonal. We need the functionality upstream so that we can all use and share, and it will not age well if no one uses. Being in a test pass or a transform or a dialect pass won't change much that equation.

I'd love if IREE and other compilers could bring more functionality to this pass and start using the upstream stuff (parametrized, cost modelled, etc). This is what makes them not age. If we don't provide a way for people to use in their projects, then MLIR gets less "batteries" (as @joker-eph usually say).